### PR TITLE
Don t encourage the usage of pre release versions

### PR DESCRIPTION
**Pull request by axbarsan**
With these changes:
* When creating a new cluster without specifying a release version, the default version used will **NOT** be a pre-release version (or a version with custom build metadata)
* When upgrading a cluster, the version picked automatically by the application will **NOT** be a pre-release version (or a version with custom build metadata)

### CODE REVIEW

File: commands/upgrade/cluster/command.go [hunk 1]
```diff
@@ -359,6 +361,15 @@ func upgradeCluster(args Arguments) (*upgradeClusterResult, error) {
 	return result, nil
 }
 
+func isVersionProductionReady(version string) bool {
+	semverVersion, err := semver.NewVersion(version)
+	if err != nil {
+		return false
+	}
+
+	return len(semverVersion.Prerelease()) < 1 && len(semverVersion.Metadata()) < 1
```
Comment (pipo02mix) [inline commands/upgrade/cluster/command.go:370]: What this `semverVersion.Metadata()` tell you?

Comment (marians) [reply commands/upgrade/cluster/command.go:370]: https://pkg.go.dev/github.com/Masterminds/semver/v3#Version.Metadata

It probably returns `-alpha1` for a string like `13.0.0-alpha1`.

Comment (marians) [reply commands/upgrade/cluster/command.go:370]: Or rather `alpha1` instead of `-alpha1`

Comment (marians) [reply commands/upgrade/cluster/command.go:370]: Well, here is an example: https://play.golang.org/p/Bbvs1yIxAn8

**Review by pipo02mix — APPROVED** [review]
LGTM

File: commands/upgrade/cluster/command.go (hunk 1 above)
Comment (axbarsan) [reply commands/upgrade/cluster/command.go:370]: In `1.0.0-alpha1+clang`, `alpha1` is the pre-release information, and `clang` is the custom build metadata.
So `.Prerelease()` would return `alpha1` and `.Metadata()` would return `clang`

**Pull request merged**